### PR TITLE
Correctly fetch `typing.Annotated` metadata in `typing.NamedTuple`

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

# TL;DR
Annotated metadata isn't included in namedtuple typing. `include_extras` should fix the issue.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Modified the import of `get_type_hints` from  `typing` to `typing_extensions` since `include_extras` is available starting Python 3.9.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by eapolinario — COMMENTED** [review]
Can you add a unit test?

**Comment by samhita-alla** [comment]
> Can you add a unit test?

Done!

**Review by pingsutw — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
Awesome!

File: flytekit/core/interface.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from collections import OrderedDict
 from typing import Any, Dict, Generator, List, Optional, Tuple, Type, TypeVar, Union
 
-from typing_extensions import get_args, get_origin
+from typing_extensions import get_args, get_origin, get_type_hints
```
Comment (wild-endeavor) [inline flytekit/core/interface.py:10]: so in python 3.7 and 3.8, it just doesn't do anything?

Comment (eapolinario) [reply flytekit/core/interface.py:10]: no, the annotations are returned regardless in python < 3.9.

Comment (eapolinario) [reply flytekit/core/interface.py:10]: I guess that's what you meant by "doing nothing".

**Comment by samhita-alla** [comment]
@pingsutw, @eapolinario, can I get a +1 again, please? Since the codecov check was still failing, I had to push a minor update.

**Review by pingsutw — APPROVED** [review]
